### DISCUSSION TITLE
multilanguage support

### DISCUSSION THREAD

**Discussion by reymalahay** [pocketbase/pocketbase#3904] [Q&A]
hi there,

first off, great framework.

aside from `golang` and `JavaScript`, will there be support for additional languages in the future? i am specifically interested in support for `Rust`, `Kotlin` and `Python`.

please advise.

thanks in advance,
rey malahay

**Comment by ganigeorgiev** [comment]
Official SDKs for interacting with the [PocketBase Web APIs](https://pocketbase.io/docs/api-records/) in other languages will be eventually implemented after stable v1 release (user contributions are also welcomed for this!), but there are no plans to add support for extending PocketBase in languages other than Go or JavaScript because it will be too difficult and time consuming for me to maintain.

Wasm integration was suggested in the past but it requires creating and maintaining a completely different set of APIs to make it work and at least for now remains out of the scope for v1. 
Note that it doesn't necessary need to be part of the core package and if someone has free time on their hands and want to experiment, it should be possible to implement it as server-side "plugin" that accept for example `core.App` value to bind into the various app hooks (eg. similar to how the current [goja "plugin"](https://github.com/pocketbase/pocketbase/blob/master/plugins/jsvm/jsvm.go) integration works), _but keep in mind that the currently exposed Go APIs are subject to change after https://github.com/pocketbase/pocketbase/discussions/3271._

**Comment by reymalahay** [comment]
@ganigeorgiev

thank you for the response. no worries.

i might contribute in the future to address this issue.

kind regards,
rey malahay


On Sat, 9 Dec 2023 at 00:23, Gani Georgiev ***@***.***> wrote:

> Official SDKs for interacting with the PocketBase Web APIs
> <https://pocketbase.io/docs/api-records/> in other languages will be
> eventually implemented after stable v1 release (user contributions are also
> welcomed for this!), but there are no plans to add support for extending
> PocketBase in languages other than Go or JavaScript because it will be too
> difficult and time consuming for me to maintain.
>
> Wasm integration was suggested in the past but it requires creating and
> maintaining a completely different set of APIs to make it work and at least
> for now remains out of the scope for v1.
> Note that it doesn't necessary need to be part of the core package and if
> someone has free time on their hands and want to experiment, it should be
> possible to implement it as server-side "plugin" that accept for example
> core.App value to bind into the various app hooks (eg. similar to how the
> current goja "plugin"
> <https://github.com/pocketbase/pocketbase/blob/master/plugins/jsvm/jsvm.go>
> integration works).
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/pocketbase/pocketbase/discussions/3904#discussioncomment-7805690>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AA2WHEIBLMVNVLED44I2OOTYIQGYBAVCNFSM6AAAAABANSXCKOVHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM3TQMBVGY4TA>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
My heroes are the ones who survived doing it wrong, who made mistakes, but
recovered from them. - Bono
